examples/map.go

```diff
@@ -0,0 +1,40 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/AlecAivazis/survey"
+)
+
+// the questions to ask
+var simpleQs = []*survey.Question{
+	{
+		Name: "name",
+		Prompt: &survey.Input{
+			Message: "What is your name?",
+		},
+		Validate: survey.Required,
+	},
+	{
+		Name: "color",
+		Prompt: &survey.Select{
+			Message: "Choose a color:",
+			Options: []string{"red", "blue", "green"},
+		},
+		Validate: survey.Required,
+	},
+}
+
+func main() {
```

Comment:
2 things...
1. I'm writing a templating CLI where a json file can be used to setup the prompts by anyone using the cli. I need to turn these into questions and get the answers so i need to build the answers container dynamically

2. I'm almost finished with another PR that you're probably not going to like, but it adds a bunch of type assertions to the copy method inside of write that can figure out the type of field on a struct and convert string values to it. e.g. you can use a struct with fields that are bool, int, float, so on and the answers will be converted to those types instead of structs only allowing string fields.

3. In the same PR, I'm adding a Converter func that's a lot like the Validator, but can be used to apply type conversion before setting the answers.